server/channels/app/platform/web_conn.go

```diff
@@ -326,6 +323,9 @@ func (wc *WebConn) SetConnectionID(id string) {
 
 // GetConnectionID returns the connection id of the connection.
 func (wc *WebConn) GetConnectionID() string {
+	if wc.connectionID.Load() == nil {
```

Comment:
A little bit unrelated, but do we _really_ need this to be an atomic? It's only set on connect and reconnect, so maybe we can save some sync costs? (Do we have tests that would detect an error on reconnect if we did remove this?) Undestandable if we don't want to introduce the chance of a race here, but I'm just thinking of ways to squeeze more perf out of this.

Reply:
Hmm .. I need to check this. Atomics are usually extremely fast, and this has never appeared in the profiles. But I am totally supportive of not using atomics unless we need to. I'll look into this separately.